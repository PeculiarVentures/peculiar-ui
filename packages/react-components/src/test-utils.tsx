import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
