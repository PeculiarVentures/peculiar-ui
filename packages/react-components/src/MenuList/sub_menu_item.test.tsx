import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { SubMenuItem } from './index';

describe('<SubMenuItem />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(<SubMenuItem>SubItem</SubMenuItem>);

    expect(asFragment()).toMatchSnapshot();
  });
});
