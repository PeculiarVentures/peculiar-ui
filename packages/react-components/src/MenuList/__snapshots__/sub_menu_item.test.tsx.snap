// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubMenuItem /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 0px var(--pv-size-base-2);
  font-family: inherit;
  outline: none;
  width: 100%;
  height: var(--pv-size-base-7);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 200ms;
  transition: background-color 200ms;
  background-color: transparent;
  border: none;
  box-sizing: border-box;
  gap: var(--pv-size-base-2);
  cursor: pointer;
  color: var(--pv-color-black);
}

.emotion-0:hover {
  background-color: var(--pv-color-gray-3);
}

.emotion-0:focus-visible,
.emotion-0[data-focused="true"] {
  background-color: var(--pv-color-gray-4);
}

.emotion-0:active,
.emotion-0[aria-selected="true"] {
  background-color: var(--pv-color-gray-5);
}

.emotion-1 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.emotion-2 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<li
    class="emotion-0"
    role="menuitem"
    tabindex="-1"
  >
    <span
      class="emotion-1"
    />
    <span
      class="emotion-2"
    >
      <svg
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m10 18 6-6-6-6"
          stroke="currentColor"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </li>
</DocumentFragment>
`;
